File: ibkr_show_frontend/src/types/adminPrompts.ts
export type PromptStatus = 'draft' | 'active' | 'archived'
export type PromptRuntimeSource = 'admin_active' | 'code_default' | 'fallback'

export interface PromptDefinition {
  prompt_key: string
  display_name: string
  module_name: string
  agent_name: string
  description: string
  default_content: string
}

export interface PromptVersion {
  id: string
  prompt_key: string
  display_name: string
  module_name: string
  agent_name: string
  description: string
  content: string
  version: string
  status: PromptStatus
  content_hash: string
  is_default: boolean
  created_at: string
  updated_at: string
  created_by: string | null
  activated_at: string | null
  change_note: string | null
}

export interface PromptListItem {
  prompt_key: string
  display_name: string
  module_name: string
  agent_name: string
  description: string
  active_version: string | null
  active_content_hash: string | null
  active_updated_at: string | null
  has_active: boolean
  is_default_active: boolean
  code_default_hash: string | null
  matches_code_default: boolean
  is_code_default_outdated: boolean
}

export interface PromptListResponse {
  items: PromptListItem[]
}

export interface PromptDetailResponse {
  definition: PromptDefinition
  versions: PromptVersion[]
  active: PromptVersion | null
}

export interface PromptRuntimeMetadata {
  prompt_key: string
  version: string | null
  content_hash: string | null
  source: PromptRuntimeSource
  error?: string
}

export interface PromptRuntimeResponse {
  content: string
  metadata: PromptRuntimeMetadata
}

export interface PromptMutationResponse {
  prompt: PromptVersion | null
  message: string
}

export interface PromptSyncCodeDefaultItem {
  prompt_key: string
  created: boolean
  skipped: boolean
  message: string
  prompt: PromptVersion | null
}

export interface PromptSyncCodeDefaultsResponse {
  created: PromptSyncCodeDefaultItem[]
  skipped: PromptSyncCodeDefaultItem[]
  message: string
}

export interface PromptCreateVersionPayload {
  content: string
  change_note?: string
}

export interface PromptActivatePayload {
  change_note?: string
}

File: ibkr_show_backend/app/schemas/admin_prompts.py
from typing import Literal

from pydantic import BaseModel, Field


class PromptDefinition(BaseModel):
    prompt_key: str
    display_name: str
    module_name: str
    agent_name: str
    description: str
    default_content: str


class PromptVersionPublic(BaseModel):
    id: str
    prompt_key: str
    display_name: str
    module_name: str
    agent_name: str
    description: str
    content: str
    version: str
    status: Literal["draft", "active", "archived"]
    content_hash: str
    is_default: bool
    created_at: str
    updated_at: str
    created_by: str | None = None
    activated_at: str | None = None
    change_note: str | None = None


class PromptListItem(BaseModel):
    prompt_key: str
    display_name: str
    module_name: str
    agent_name: str
    description: str
    active_version: str | None = None
    active_content_hash: str | None = None
    active_updated_at: str | None = None
    has_active: bool
    is_default_active: bool = False
    code_default_hash: str | None = None
    matches_code_default: bool = False
    is_code_default_outdated: bool = False


class PromptCreateVersionRequest(BaseModel):
    content: str = Field(min_length=1)
    change_note: str | None = None


class PromptActivateRequest(BaseModel):
    change_note: str | None = None


class PromptRuntimeMetadata(BaseModel):
    prompt_key: str
    version: str | None = None
    content_hash: str | None = None
    source: str


class PromptRuntimeResponse(BaseModel):
    content: str
    metadata: PromptRuntimeMetadata


class PromptDetailResponse(BaseModel):
    definition: PromptDefinition
    versions: list[PromptVersionPublic]
    active: PromptVersionPublic | None = None


class PromptListResponse(BaseModel):
    items: list[PromptListItem]


class PromptMutationResponse(BaseModel):
    prompt: PromptVersionPublic | None = None
    message: str


class PromptSyncCodeDefaultItem(BaseModel):
    prompt_key: str
    created: bool
    skipped: bool
    message: str
    prompt: PromptVersionPublic | None = None


class PromptSyncCodeDefaultsResponse(BaseModel):
    created: list[PromptSyncCodeDefaultItem]
    skipped: list[PromptSyncCodeDefaultItem]
    message: str
